op-chain-ops/crossdomain/message.go

```diff
@@ -67,6 +68,26 @@ func (c *CrossDomainMessage) Hash() (common.Hash, error) {
 	case 1:
 		return HashCrossDomainMessageV1(c.Nonce, c.Sender, c.Target, c.Value, c.GasLimit, c.Data)
 	default:
-		return common.Hash{}, fmt.Errorf("unknown nonce version %d", version)
+		return common.Hash{}, fmt.Errorf("unknown version %d", version)
+	}
+}
+
+// ToWithdrawal will turn a CrossDomainMessage into a Withdrawal.
```

Comment:
I'm still a little unclear about what this function is supposed to do

Reply:
V0 cross domain messages have all of the information required to create the corresponding withdrawal type, so this is a helper to get access to the the methods on the withdrawal given the same data in the cross domain message

Reply:
Oh ok I see, thanks for the clarification!